### DISCUSSION TITLE
Why is the rule applied against the caller? (new-cap)

### DISCUSSION THREAD

**Discussion by nedsociety** [eslint/eslint#15722] [Q&A] [answered]
I'm fairly new to frontend dev in general so bear with me if I'm missing any obvious point.

I've setup eslint with google style and found the flagged code with `new-cap`:

![image](https://user-images.githubusercontent.com/11943013/159839986-352ef9e1-c421-42e5-b04c-fcb6129aecd1.png)

Well the notion makes sense, eslint wants it to be camelCased. But that's [3rd party code](https://github.com/weichx/cerialize#details) and I don't really have a good fix against it:

1. I don't want to disable `new-cap` rule globally because I'm willing to enforce that rule in _my_ code.
2. I don't want to make exception clause for every call on that function, because it feels pretty dumb: `// eslint-disable-next-line`
3. Nor to alias every import for the same reason: `import {Deserialize as deserializeFunc} from 'cerialize';`

The fundamental question here is, why is the rule checked against the caller? My code doesn't seem to be blamed, it's the callee not following that naming rule. Of course it's not even their responsibility because the author of that library might be using other coding style!

Is eslint designed to force me to ban libraries that doesn't have the same style? Or is there any fix that I'm missing?

**Comment by nzakas** [comment, marked as answer]
This rule is basically copied from JSHint for compatibility with existing style guides. This came back from before ESM, so there was no way to tell if something was library code or not.

If you open a bug for this, I think we can change it so imports are ignored.

**Reply by nedsociety** [reply]
Thanks. #15724
